Explain trim behavior while rebuilding and clean up doc (#868)
longhorn/website: content/docs/1.7.0/nodes-and-volumes/volumes/trim-filesystem.md

## Automatically Remove Snapshots During Filesystem Trim

By design each valid snapshot of a Longhorn volume is immutable. Hence, the Longhorn filesystem trim feature can be
applied to **the volume head and the preceding continuously removed or system snapshots only**. If most of the actual

> space consumed by a volume is associated with valid snapshots, the trim operation will not be very effective.

Reviewer: I'm not sure if I understood the bolded text correctly. If I did, an unordered list might be more effective.

> space consumed by a volume is associated with valid snapshots, the trim operation will not be very effective.

Author: This one's a bit tricky. I think we need to associate continuous chain with both removed and system snapshots. I half adopted your suggestion. WDYT.

### The Global Setting "Remove Snapshots During Filesystem Trim"

To help reclaim as much space as possible automatically, Longhorn includes the setting
[_Remove Snapshots During Filesystem Trim_](../../../references/settings/#remove-snapshots-during-filesystem-trim).
This allows the trim feature to automatically mark the latest snapshot and all preceding snapshots
(until there is a fork in the snapshot chain) as removed. As a result, Longhorn can reclaim space for as many snapshots
as possible. However, this setting also causes intentionally created snapshots to be marked as removed (and eventually
purged), so it should be used with caution.

### The Volume Spec Field "UnmapMarkSnapChainAsRemoved"

There is a per-volume field `volume.Spec.UnmapMarkSnapChainAsRemoved` that overwrites the global setting mentioned above.

There are 3 options for this volume field: `ignored`, `enabled`, and `disabled`. `ignored` means followi the global
setting, which is the default value.

You can directly set this field in a StoragaClass so that the customized value can be applied to all volumes created by
the StorageClass.

## Known Issues & Limitations

> ### Rebuilding volumes

Reviewer: ```suggestion
### Rebuilding Volumes
```

> The trim operation cannot be executed successfully against a volume that is actively rebuilding. By design, it unmaps
> blocks in the volume head and all the continuously removed snapshots preceding it. However, some of these snapshots may
> be actively transferring from one replica to another while rebuilding is ongoing.

Reviewer: ```suggestion
By design, Longhorn unmaps blocks in the volume head and in the preceding continuous chain of snapshots marked as removed. Some of these snapshots may be moved from one replica to another during volume rebuilding, so Longhorn is unable to trim the filesystem of affected volumes when rebuilding is in progress.
```

> Instead of returning an I/O error to the filesystem, Longhorn silently refuses to unmap blocks during a rebuild. The
> rebuild may take a long time, and VM workloads in particular respond poorly when repeated attempts to complete a trim
> return errors. See https://github.com/longhorn/longhorn/issues/7103 for details.

Reviewer: ```suggestion
Instead of returning an I/O error to the filesystem, Longhorn simply does not unmap blocks during rebuilding. Consequently, VM workloads respond poorly when attempts to complete the trim operation return errors, and rebuilding may take longer than expected. For more information, see [Issue #7103](https://github.com/longhorn/longhorn/issues/7103).
```

Author: This one changes my intended meaning. Please check my new wording.

> If a trim operation is started during a rebuild, it will have no effect. Similar to issuing a trim against a valid
> snapshot, the filesystem will remember that it has already trimmed the associated blocks. Later on, if you want to
> attempt the trim again (to recover space), you may need to unmount and remount the filesystem first.

Reviewer: ```suggestion
A trim operation that is started during rebuilding has no effect because the filesystem remembers whenever it has trimmed associated blocks. You may need to unmount and remount the filesystem before attempting to start the trim operation again.
```

Author: This one also changes the intended meaning. Please check my new wording.

> ### Expanding volumes

Reviewer: ```suggestion
### Expanding Volumes
```

> The trim operation cannot be executed successfully against a volume that is actively expanding. Unlike for rebuilding
> volumes, Longhorn returns an I/O error to the filesystem when this occurs. Expansion is generally fast. It is better
> to inform the filesystem that the trim failed so that it can be tried again soon (without remounting).

Reviewer: ```suggestion
Longhorn is unable to trim the filesystem during volume expansion and returns an I/O error whenever the issue is encountered. Consider informing the filesystem about any unsuccessful trim operation so that another attempt can be made without the need for remounting.
```

Author: This one also changes the intended meaning. Please check my new wording.

> ### Encrypted volumes

Reviewer: ```suggestion
### Encrypted Volumes
```
